server/application-server/src/main/java/de/tum/cit/aet/helios/environment/EnvironmentController.java

```diff
@@ -74,7 +74,18 @@ public ResponseEntity<?> unlockEnvironment(@PathVariable Long id) {
   @PutMapping("/{id}/lock")
   public ResponseEntity<?> lockEnvironment(@PathVariable Long id) {
     Optional<Environment> environment = environmentService.lockEnvironment(id);
-    return ResponseEntity.ok(EnvironmentDto.fromEnvironment(environment.get()));
+    return environment
+        .map(env -> ResponseEntity.ok(EnvironmentDto.fromEnvironment(env)))
+        .orElse(ResponseEntity.status(403).build());
+  }
+
+  @EnforceAtLeastWritePermission
+  @PutMapping("/{id}/extend-lock")
+  public ResponseEntity<?> extendEnvironmentLock(@PathVariable Long id) {
+    Optional<Environment> environment = environmentService.extendEnvironmentLock(id);
+    return environment
+        .map(env -> ResponseEntity.ok(EnvironmentDto.fromEnvironment(env)))
+        .orElse(ResponseEntity.status(403).build());
```

Comment:
issue: can we throw a EntityNotFoundException with a message instead? This way we miss the client notification for the user on an error right?